graphql_compiler/compiler/emit_sql.py

```diff
@@ -205,25 +289,66 @@ def _set_group_by(self, group_by):
         """Set output columns for the fold object."""
         self._group_by = group_by
 
-    def _construct_fold_joins(self, edge, from_alias, to_alias):
+    def _construct_fold_joins(self):
         """Use the edge descriptors to create the join clause between the tables in the fold."""
-        join_clause = sqlalchemy.join(
-            from_alias,
-            to_alias,
-            onclause=(from_alias.c[edge.from_column] == to_alias.c[edge.to_column])
-        )
+        # Start the join clause with the from_table of the first join descriptor
+        _, join_clause, _ = self.join_info[0]
+
+        # Starting at the first from_table, join traversed vertices until the output vertex is
+        # reached.
+        #
+        # MSSQL doesn't use the JOIN for the very last traversal,
+        # so we pop it off before construction the JOIN clause
+        if self._dialect == 'mssql':
+            # Move the final join predicate (from the vertex preceding the output to
+            # the output vertex) into the WHERE clause of the SELECT ... FOR XML PATH('')
+            self.join_info.pop()
+        while len(self.join_info) > 0:
+            # joins from earlier in the chain of traversals are at the end of the list
+            # b/c joins are appended in the order they are traversed
+            edge, from_alias, to_alias = self.join_info.pop(0)
```

Comment:
Here we seem to be popping from the front of the list, which makes this loop unnecessarily be `O(n^2)`. Can't we just iterate over the list in order to construct the `join_clause` object and then just set the reference to an empty list? Isn't that `O(n)`?